### PR TITLE
feat: add Datadog RUM to template and components

### PR DESCRIPTION
**Pull request by dcshzj**
## Problem

We don't have RUM set up on our template, so we are not able to monitor the user behaviour on the actual Isomer Next sites.

## Solution

**Breaking Changes**

- [ ] Yes - this PR contains breaking changes
- [x] No - this PR is backwards compatible

**Features**:

- Set up Datadog RUM monitoring as a component, and use the component inside the Skeleton layout.
- The sampling rate is configured using the environment variable `NEXT_PUBLIC_ISOMER_NEXT_RUM_SAMPLING_RATE`, which is injected at build time to feature flag the level of sampling.

### CODE REVIEW

File: packages/components/src/templates/next/components/internal/DatadogRum/DatadogRum.tsx [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    process.env.NEXT_PUBLIC_ISOMER_NEXT_RUM_SAMPLING_RATE ?? 0,
+  ),
+  sessionReplaySampleRate: Number(
+    process.env.NEXT_PUBLIC_ISOMER_NEXT_RUM_SAMPLING_RATE ?? 0,
+  ),
+  trackUserInteractions: true,
+  trackResources: true,
+  trackLongTasks: true,
+  defaultPrivacyLevel: "allow",
+})
```
Comment (adriangohjw) [inline packages/components/src/templates/next/components/internal/DatadogRum/DatadogRum.tsx:24]: should we pass in `production` as ENV too? or check that its production/staging before rendering. so as to not dirty the data + not have it run on dev and incur unncessary cost

File: apps/studio/public/assets/css/preview-tw.css
Comment (adriangohjw) [inline apps/studio/public/assets/css/preview-tw.css:1]: ooo why this file will change ah?

**Review by adriangohjw — APPROVED** [review]

File: packages/components/src/templates/next/components/internal/DatadogRum/DatadogRum.tsx (hunk 1 above)
Comment (dcshzj) [reply packages/components/src/templates/next/components/internal/DatadogRum/DatadogRum.tsx:24]: Yup, the env is provided earlier on line 13, but I also added a check to only render this component if the site is not on staging

File: apps/studio/public/assets/css/preview-tw.css
Comment (dcshzj) [reply apps/studio/public/assets/css/preview-tw.css:1]: This one is an automatic update on build

**Pull request merged**